### ISSUE TITLE
OAuth2 Support

### ISSUE THREAD

**Issue by sahbi-ktifa** [jhipster/jhipster-vuejs#203]
##### **Overview of the feature request**

OAuth2 authentication mechanism should be supported as well as JWT and Session.

##### **Motivation for or Use Case**

Having all three authentication mechanisms available.

-   [ ] Checking this box is mandatory (this is just to show you read everything)

**Comment by VinodAnandan** [comment]
https://www.npmjs.com/package/vuex-oidc

https://github.com/perarnborg/vuex-oidc/wiki

**Comment by atomfrede** [comment]
@jdubois @jhipster/vuejs-developers I think this could be worth a bounty, WDYT?

**Comment by sahbi-ktifa** [comment]
Yes, it might be eligible. @atomfrede did you start to work on this one ? If not, I'd like to give it try.

**Comment by atomfrede** [comment]
@sahbi-ktifa No I didn't start, so go ahead!

**Comment by sahbi-ktifa** [comment]
Bounty claimed https://opencollective.com/generator-jhipster/expenses/6650
